source/verus/src/main.rs

```diff
@@ -75,19 +63,270 @@ fn main() {
+    std::process::Command::new("git")
+        .current_dir(&proj_path)
+        .arg("commit")
+        .arg("-m")
+        .arg("\"verus\"")
+        .env("GIT_AUTHOR_NAME", "nobody")
+        .env("GIT_AUTHOR_EMAIL", "nobody@nobody.nobody")
+        .env("GIT_COMMITTER_NAME", "nobody")
+        .env("GIT_COMMITTER_EMAIL", "nobody@nobody.nobody")
+        .output()
+        .map_err(|x| format!("localstorage: failed to git commit with error message: {}", x))?;
+
+    Ok(())
+}
+
+#[derive(Debug)]
+pub struct Reports {
+    proj_path: PathBuf,
+    dep_path: PathBuf,
+}
+
+fn repo_path(store: bool) -> Option<Reports> {
+    // check if user has git as executable
+    if Command::new("git").arg("--version").output().is_err() {
+        return None;
+    }
+
+    // check if user has local storage enabled during vargo build
+    if option_env!("VERUS_LOCAL_STORE").is_none() {
+        return None;
+    }
+
+    // Check if there's a verus/reports/.git file in the user's local data lib
+    // if not so, create verus/ directory
+    let reports_dir = dirs::data_local_dir()?.join("verus").join("reports");
+
+    if !reports_dir.join(".git").is_file() {
+        create_dir_all(&reports_dir).expect("Creating verus/ directory in local data dir");
+
+        Command::new("git")
+            .current_dir(&reports_dir)
+            .arg("init")
+            .output()
+            .expect("Initializing git in local data dir");
+    }
+
+    let uuid_file = reports_dir.join("userid");
+    if !uuid_file.is_file() {
+        let uuid = uuid::Uuid::new_v4();
+        write(&uuid_file, uuid.to_string()).expect("Writing user id file");
+    }
+
+    let uuid = read_to_string(&uuid_file).expect("Reading user id file");
+
+    // Project dir check/create (project = SHA256((normalized/absolute)path to root.rs)
+    let new_args = std::env::args().into_iter();
+
+    let rs_files = new_args.filter(|arg| arg.ends_with(".rs")).collect::<Vec<_>>();
+
+    let input_file: Option<String> = match rs_files.len() {
+        1 => Some(rs_files[0].clone()),
+        _ => {
+            None // passed down to rust_verify for error message
+        }
+    };
+
+    let input_file_path = input_file.and_then(|f| std::fs::canonicalize(f).ok());
+
+    let project_name = input_file_path
+        .as_ref()
+        .map(|path| format!("{:x}", sha2::Sha256::digest(path.to_str().unwrap().as_bytes())));
+    // not sure how to write a ? to replace unwrap
+
+    let proj_dir = project_name.map(|name| {
+        let project_dir = reports_dir.join(uuid).join(name);
+        if !project_dir.is_dir() && store {
+            create_dir_all(&project_dir).expect("failed to create project directory");
+        }
+        project_dir
+    });
+
+    let dir = std::env::temp_dir();
+    let temp_file = dir.join("verus-dep-info");
+
+    match (input_file_path, proj_dir) {
+        (Some(_), Some(proj_dir)) => Some(Reports { proj_path: proj_dir, dep_path: temp_file }),
+        _ => None,
+    }
```

Comment:
Why is this not handled via `?`s when you make `input_file_path` and `proj_dir`?